### PR TITLE
Reduces BasicErrorHandler's complexity

### PR DESCRIPTION
**Pull request by funkyfuture**
relates to https://github.com/pyeve/cerberus/pull/321/files#r124809777 ff

### CODE REVIEW

**Comment by funkyfuture** [comment]
cc @kornholi, please have a look.

(@nicolaiarocci, can you enable me to request reviews by other users?)

**Comment by funkyfuture** [comment]
this should be rebased after #344 / 331c56b is merged to test the docs for updated representations.

File: cerberus/errors.py [hunk 1]
```diff
@@ -525,89 +535,76 @@ def insert_error(self, path, node):
...
             child_error.document_path = error.document_path + relative_path
 
-            self.rewrite_error_path(child_error, offset)
+            self._rewrite_error_path(child_error, offset)
 
-    def rewrite_logic_error_path(self, error, offset=0):
-        error.document_path = error.document_path + (error.rule, )
-        child_start = len(error.document_path) - offset - 1
+    def _rewrite_logic_error_path(self, error, offset=0):
+        error.document_path = error.document_path
```
Comment (kornholi) [inline cerberus/errors.py]: 😄

File: cerberus/errors.py [hunk 2]
```diff
@@ -461,7 +461,7 @@ class BasicErrorHandler(BaseErrorHandler):
 
                 0x91: "one or more definitions validate",
                 0x92: "none or more than one rule validate",
-                0x93: "no definitions validate",
+                0x93: "not any definition validates",
```
Comment (kornholi) [inline cerberus/errors.py]: these have the same meaning, but the new one sounds more awkward

**Comment by kornholi** [comment]
thanks for doing this. looks good except for the changed error message - what was your reasoning?

File: cerberus/errors.py (hunk 2 above)
Comment (funkyfuture) [reply cerberus/errors.py]: i wanted to include a reference to the rule that evokes this message. do you have a proposal?

File: cerberus/errors.py (hunk 1 above)
Comment (funkyfuture) [reply cerberus/errors.py]: :-D looks like a missing feature for `flake8`.

**Comment by nicolaiarocci** [comment]
Is this ready for merge?

**Comment by funkyfuture** [comment]
nah, i'd like to rebase it and have a consensus on the wording that @kornholi addressed.

**Comment by funkyfuture** [comment]
okay, it's rebased. i also reverted to the prior message text. the information is given implititly later and i don't care enough to thnk about something better.

**Comment by nicolaiarocci** [comment]
Rebased and merged

**Pull request closed without merging**